src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -209,76 +233,107 @@ export const GenericTable = ({
+    const query = dynamicValuesConfig?.query;
+    if (!query) {
+      setPlaceholderTitle(t('genericcomponent.table.labels.noQueryPlaceholderTitle', 'No data'));
+      setPlaceholderBody(
+        t(
+          'genericcomponent.table.labels.NoQueryErrorImportFalse',
+          'Impossible to fetch data from dataset because there is no twingraph query defined for this parameter. '
+        )
+      );
+      return;
+    }
+
+    const resultKey = dynamicValuesConfig?.resultKey;
+    let data;
     try {
-      if (!currentScenarioDatasetList || currentScenarioDatasetList?.length === 0)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.noDatasetsError',
-            'Impossible to fetch data from dataset because the list of datasets is empty'
-          )
-        );
-      const sourceDatasetId = currentScenarioDatasetList[0];
-      const dynamicValuesConfig = ConfigUtils.getParameterAttribute(parameterData, 'dynamicValues');
-      const query = dynamicValuesConfig?.query;
-      if (!query)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.noQueryError',
-            'Impossible to fetch data from dataset because there is no twingraph query defined for this parameter. ' +
-              'You can load data manually using Import button'
-          )
-        );
-      const resultKey = dynamicValuesConfig?.resultKey;
-      const { data } = await Api.Datasets.twingraphQuery(organizationId, sourceDatasetId, { query });
-      if (data.length === 0)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.wrongQueryError',
-            'Returned result is empty, there is probably an error in your query configuration. ' +
-              'Please, check your solution'
-          )
-        );
-      const rowsDict = data.map((row) => row[resultKey]);
-      if (rowsDict.includes(undefined))
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.wrongResultKeyError',
-            `Returned result doesn't have ${resultKey} property.
-      Probably there is an error in resultKey configuration, please, check your solution`,
-            { resultKey }
-          )
-        );
-      const csvRows = AgGridUtils.toCSV(rowsDict, columns);
-      const agGridData = _generateGridDataFromCSV(csvRows, parameterData);
-      if (agGridData.error) {
-        setClientFileDescriptor({
-          tableDataStatus: TABLE_DATA_STATUS.ERROR,
-          errors: agGridData.error,
-        });
-      } else
-        setClientFileDescriptor({
-          name: fileName,
-          file: null,
-          agGridRows: agGridData.rows,
-          errors: null,
-          status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
-          tableDataStatus: TABLE_DATA_STATUS.READY,
-          uploadPreprocess: { content: _uploadPreprocess },
-        });
+      ({ data } = await Api.Datasets.twingraphQuery(organizationId, sourceDatasetId, { query }));
```

Comment:
There are parenthesis around the api call, I don't know this syntax, is it some linter games or something new ? :slightly_smiling_face: